defi-pulse/src/data-providers/coinpaprika.ts

```diff
@@ -0,0 +1,43 @@
+import { Requester } from '@chainlink/external-adapter'
+import { Index } from '../adapter'
+
+const getPriceData = async (currency: string) => {
+  const url = 'https://api.coinpaprika.com/v1/tickers'
+  const params = {
+    quotes: currency.toUpperCase(),
+  }
+  const config = {
+    url,
+    params,
+  }
+  const response = await Requester.request(config)
+  return response.data
+}
+
+const toAssetPrice = (data: Record<string, any>, currency: string) => {
+  const price = data.quotes && data.quotes[currency.toUpperCase()].price
+  if (!price || price <= 0) {
+    throw new Error('invalid price')
+  }
+  return price
+}
+
+const getPriceIndex = async (index: Index, currency: string): Promise<Index> => {
+  const priceData = await getPriceData(currency)
+
+  const sortedData = priceData.sort((a: any, b: any) => a.rank - b.rank)
```

Comment:
Maybe just add a quick comment here that we remove duplicates after sorting by rank.